app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,187 @@
+        else
+          ssaq_args = {}
+          ssaq_args[:ssa_bucket]    = coord.ssa_bucket
+          ssaq_args[:reply_queue]   = coord.reply_queue
+          ssaq_args[:request_queue] = coord.request_queue
+          ssaq_args[:region]        = coord.ems.provider_region
+          ssaq_args[:sqs]           = coord.sqs
+          ssaq_args[:s3]            = coord.s3
+          ssaq                      = AmazonSsaSupport::SsaQueue.new(ssaq_args)
+          unless ssaq
+            _log.error("Error creating SsaQueue for #{coord.ems.name}")
+            next
+          end
+          _log.debug("Getting replies for #{coord.ems.name}")
+          begin
+            ssaq.reply_loop do |reply|
+              _log.debug("Reply for #{coord.ems.name}: #{reply}")
+              perform_metadata_sync(reply)
+            end
+          rescue => err
+            _log.error("Error #{err} processing replies for #{coord.ems.name}.  Continuing.")
+            next
+          end
+        end
+      end
+      response_check_sleep_seconds = response_thread_sleep_seconds
+      _log.debug("going to sleep for #{response_check_sleep_seconds} seconds after checking all Agent Coordinators")
+      sleep(response_check_sleep_seconds) unless @shutdown_instance_wait_thread
+    end
+  end
+
+  def perform_metadata_sync(extract_reply)
+    ost       = OpenStruct.new
+    ost.reply = extract_reply
+    ost.jobid = extract_reply[:job_id]
+    job       = Job.find_by(:id => ost.jobid)
+    raise _("Unable to sync data for job with id #{ost.jobid}") if job.nil?
+    target_id  = job.target_id
+    vm         = VmOrTemplate.find(target_id)
+    ost.taskid = job.guid
+    unless vm.kind_of?(ManageIQ::Providers::Amazon::CloudManager::Vm) ||
+           vm.kind_of?(ManageIQ::Providers::Amazon::CloudManager::Template)
+      error = "Vm #{vm.name} of class #{vm.class.name} is not an Amazon instance or image - unable to sync data"
+      update_job_message(ost, error)
+      job.signal(:abort, error, "error")
+      return
+    end
+    _log.debug("Syncing Metadata for #{vm.ems_ref}")
+    if extract_reply[:error]
+      update_job_message(ost, extract_reply[:error])
+      job.signal(:abort, extract_reply[:error], "error")
+      return
+    end
+    update_job_message(ost, "Synchronization in progress")
+    status        = scan_message         = "OK"
+    status_code   = categories_processed = 0
+    ost.xml_class = XmlHash::Document
+    bb            = last_err = nil
+
+    xml_summary = ost.xml_class.createDoc(:summary)
+    xml_node    = xml_node_scan = xml_summary.root.add_element("scanmetadata")
+    xml_node_scan.add_attributes("start_time" => extract_reply[:start_time])
+    xml_summary.root.add_attributes("taskid" => ost.taskid)
+    ost.taskid = ost.jobid
+
+    data_dir = File.expand_path(Rails.root.join("data", "metadata"))
+    _log.debug("creating #{data_dir}")
+    unless File.exist?(data_dir)
+      begin
+        Dir.mkdir(data_dir)
```

Comment:
You probably want `FileUtils.mkdir_p` instead

Reply:
Fixed in https://github.com/ManageIQ/manageiq-providers-amazon/pull/360